presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -735,9 +739,23 @@ public synchronized Map<String, Optional<Partition>> getPartitionsByNames(Metast
         }
         Map<String, Optional<Partition>> delegateResult = delegate.getPartitionsByNames(metastoreContext, databaseName, tableName, partitionNamesToQuery.build());
         resultBuilder.putAll(delegateResult);
+
+        cacheLastAccessTimes(delegateResult, databaseName, tableName);
```

Comment:
if this information is already in the partition, why do we need to cache it separately?

Reply:
These time information will be returned during the commit phase. Do you know if we already cache partition information somewhere? If so, I can collect these time information during the commit phase.

Reply:
I was thinking of partitionActions, but i guess that doesn't have the input information.